client/src/components/ResourceHub/AddResourceModal.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { ReactElement, FC, Dispatch, useState } from 'react';
+import { Modal, Button, Form, Input, Checkbox } from 'semantic-ui-react';
+
+import { createResource, isError } from '../../utils/apiWrapper';
+
+import '../../css/AddResourceModal.css';
+
+const teams = [
+  'General',
+  'Editing',
+  'Factchecking',
+  'Illustration',
+  'Photography',
+  'Onboarding',
+  'Visuals',
+  'Writing',
+];
+
+interface IProps {
+  onAdd: Dispatch<void>;
+}
+
+const AddResourceModal: FC<IProps> = ({ onAdd }): ReactElement => {
+  const [open, setOpen] = useState<boolean>(false);
+  const [resourceTitle, setResourceTitle] = useState<string>('');
+  const [resourceURL, setResourceURL] = useState<string>('');
+  const [selectedTags, setSelectedTags] = useState<Set<string>>(new Set());
+
+  const handleTagSelect = (tag: string): void => {
+    const newTags = new Set(selectedTags);
+    if (newTags.has(tag)) {
```

Comment:
You can make use of the ternary operator here: `newTags.has(tag) ? newTags.delete(tag) : newTags.add(tag);`